av1an-cli/src/lib.rs

```diff
@@ -442,164 +442,217 @@ fn confirm(prompt: &str) -> io::Result<bool> {
   }
 }
 
-pub fn parse_cli(args: CliOpts) -> anyhow::Result<EncodeArgs> {
-  let temp = if let Some(path) = args.temp.as_ref() {
-    path.to_str().unwrap().to_owned()
+/// Given Folder and File path as inputs
+/// Converts them all to file paths
+/// Converting only depth 1 of Folder paths
+pub fn convert_input(path: &Path) -> anyhow::Result<Vec<PathBuf>> {
+  // TODO: to validate file extensions
+  // let valid_media_extensions = ["mkv", "mov", "mp4", "webm", "avi", "qt", "ts", "m2t", "py", "vpy"];
+
+  if path.is_file() {
+    Ok(vec![path.to_path_buf()])
+  } else if path.is_dir() {
+    Ok({
+      let dir = std::fs::read_dir(path)?;
+      dir
```

Comment:
Since this also exists in other places in the code, I think this should be its own function.